sabnzbd/scheduler.py

```diff
@@ -349,6 +351,32 @@ def plan_resume(self, interval):
             self.pause_end = None
             sabnzbd.unpause_all()
 
+    def __check_diskspace(self, full_dir, required_space):
+        """ Resume if there is sufficient available space """
+        disk_free = diskspace(force=True)[full_dir][1]
+        if disk_free > required_space:
+            logging.debug("Resuming, %s has %d GB free, needed %d GB", full_dir, disk_free, required_space)
+            sabnzbd.unpause_all()
+            self.scheduler.cancel(self.resume_task)
+            self.resume_task = None
+        else:
+            logging.debug("%s has %d GB free, need %d GB to resume", full_dir, disk_free, required_space)
+
+    def plan_diskspace_resume(self, full_dir=None, required_space=None):
+        """ Create regular check for free disk space """
+        self.cancel_resume_task()
+        logging.debug("Pausing, will resume when %s has more than %d GB free", full_dir, required_space)
+        self.resume_task = self.scheduler.add_interval_task(
+            self.__check_diskspace, "check_diskspace", 5 * 60, 9 * 60, "threaded", args=[full_dir, required_space]
+        )
+        sabnzbd.Downloader.pause()
+
+    def cancel_resume_task(self):
```

Comment:
I think it's cleaner to remove the tasks that are not wanted than to make logic to ignore them. For instance, if it was resumed manually and then paused again manually it shouldn't be resumed by a lingering disk check task. If other kinds of resume tasks are added later it can get hard to keep track.

With the current task in `self.resume_task` it's easy to check the existence or cancel it. You can know which type it is by checking `self.resume_task.name` or the next time it will be executed through `self.resume_task.event.time`.

Reply:
That's indeed a good reason 👍